java/src/test/java/com/saucelabs/simplesauce/SauceSessionTest.java

```diff
@@ -122,14 +109,52 @@ public void sauceOptions_startWithChrome_startsChrome() {
 
     @Test(expected = NullPointerException.class)
     public void stop_newWebDriverInstanceSetByStart_stopsSession() {
-        sauce = new SauceSession(dummyRemoteDriver, dummyEnvironmentManager);
+        sauce = new SauceSession(dummyRemoteDriver);
 
         WebDriver driver = sauce.start();
         sauce.stop();
 
         driver.quit();
     }
 
+    @Test
+    public void usesENVAsDefaultUsername() {
+        sauce = Mockito.spy(new SauceSession(options, dummyRemoteDriver));
+        doReturn("test-name").when(sauce).getEnvironmentVariable("SAUCE_USERNAME");
+    }
+
+    @Test
+    public void usesENVAsDefaultAccessKey() {
+        sauce = Mockito.spy(new SauceSession(options, dummyRemoteDriver));
+        doReturn("accesskey").when(sauce).getEnvironmentVariable("SAUCE_ACCESS_KEY");
+    }
+
+    @Test
+    public void startThrowsErrorWithoutUsername() {
+        sauce = Mockito.spy(new SauceSession(options, dummyRemoteDriver));
+        doReturn(null).when(sauce).getEnvironmentVariable("SAUCE_ACCESS_KEY");
+
+        try {
+            sauce.start();
+            fail("Expected a SauceEnvironmentVariablesNotSetException to be thrown");
+        } catch (SauceEnvironmentVariablesNotSetException exception) {
+            assertEquals("Sauce Access Key was not provided", exception.getMessage());
+        }
+    }
+
+    @Test
+    public void startThrowsErrorWithoutAccessKey() {
+        sauce = Mockito.spy(new SauceSession(options, dummyRemoteDriver));
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Unnecessary use of fully qualified name 'Mockito.spy' due to existing static import 'org.mockito.Mockito.*'](https://app.codacy.com/app/SauceLabs/simple_sauce/pullRequest?prid=4732268)